ITISFoundation/osparc-simcore
# ✨ Implementing new `list_services_paginated` entrypoint in web-api (part 1)

## What do these changes do?

In PR https://github.com/ITISFoundation/osparc-simcore/pull/6003 we implemented a new `rpc` interface in the `catalog` and connected it with the `rest` api of the `webserver` service. This PR **partially** implements `api.rpc._services.list_services_paginated`. 

Specifically:
- implements repository layer (i.e. `db.repositories.services`)
- implements services layer (i.e. `services.catalog.list_services_paginated`)
     -  only the data from the database is included.
     -  data from image labels is still not included
- connects controller layer (i.e. `api.rpc._services`) to service layer 
- ♻️ Adapting `ServiceGetV2` model in coordination with @odeimaiz 

Following up PRs will cover:
- connection with registry/director
- compatibility rules


## Related issue/s

- Follows up from https://github.com/ITISFoundation/osparc-simcore/pull/6003
- Part of #5964 


## How to test

Driving tests
- services/catalog/tests/unit/with_dbs/test_db_repositories.py
- Manual exploratory test: use `/dev/catalog/services-`
  -  set `WEBSERVER_DEV_FEATURE_ENABLED=1` (i.e. configuration in master deploy)
  - start stack
  - open swagger at `/dev/doc`
  - execute `GET /dev/catalog/services/-/latest`. Expected fast and short response body
  - compare with `GET /catalog/services/latest`. Expected slow and long response body

## Dev-ops checklist
None
